tests/php/Forms/NumericFieldTest.php

```diff
@@ -127,6 +127,19 @@ public function testNumberTypeOnInputHtml()
         $this->assertContains('type="text"', $html, 'number type not set');
     }
 
+    public function testNullSet() {
+        $field = new NumericField('Number');
+        $field->setValue('');
+        $this->assertNull($field->Value());
```

Comment:
It's coming up as `''`, which is probably should be the expected value for this test. Only the datavalue() should be null, although it emits an empty string to the user through the front-end Value() method

I suggest:
 - changing setValue() to setSubmittedValue() (to simulate a user submitting an empty string)
 - expect these invalid strings to be returned as-is ($this->originalValue).